publication-rest/src/main/java/no/unit/nva/publication/validation/ETag.java

```diff
@@ -0,0 +1,71 @@
+package no.unit.nva.publication.validation;
+
+import static java.util.Objects.isNull;
+import static nva.commons.core.StringUtils.isBlank;
+import java.util.Objects;
+import nva.commons.core.JacocoGenerated;
+
+/**
+ * Represents an ETag for HTTP caching and validation based on entity version and user context.
+ *
+ * <p>ETags are used to validate cached responses. This implementation combines entity version
+ * with username to ensure cache invalidation when the user's authorization state changes.
+ *
+ * <p>Format: {@code username:version} or {@code version} (when username is null)
+ */
+public final class ETag {
```

Comment:
Could probably be a record?

Reply:
Record has default constructor. I want to enforce validation of values via factory methods. I can also enforce it via custom record constructor, but it will be code duplication.